java/flight/flight-integration-tests/src/main/java/org/apache/arrow/flight/integration/tests/FlightSqlScenarioProducer.java

```diff
@@ -0,0 +1,343 @@
+                                                   CallContext context,
+                                                   FlightDescriptor descriptor) {
+    IntegrationAssertions.assertEquals(command.getPreparedStatementHandle().toStringUtf8(),
+        "SELECT PREPARED STATEMENT HANDLE");
+
+    return getFlightInfoForSchema(command, descriptor, getQuerySchema());
+  }
+
+  @Override
+  public SchemaResult getSchemaStatement(FlightSql.CommandStatementQuery command,
+                                         CallContext context, FlightDescriptor descriptor) {
+    return new SchemaResult(getQuerySchema());
+  }
+
+  @Override
+  public void getStreamStatement(FlightSql.TicketStatementQuery ticket, CallContext context,
+                                 ServerStreamListener listener) {
+    serveJsonToStreamListener(listener, getQuerySchema());
+  }
+
+  @Override
+  public void getStreamPreparedStatement(FlightSql.CommandPreparedStatementQuery command,
+                                         CallContext context, ServerStreamListener listener) {
+    serveJsonToStreamListener(listener, getQuerySchema());
+  }
+
+  private Runnable acceptPutReturnConstant(StreamListener<PutResult> ackStream, int value) {
+    return () -> {
+      final FlightSql.DoPutUpdateResult build =
+          FlightSql.DoPutUpdateResult.newBuilder().setRecordCount(value).build();
+
+      try (final ArrowBuf buffer = allocator.buffer(build.getSerializedSize())) {
+        buffer.writeBytes(build.toByteArray());
+        ackStream.onNext(PutResult.metadata(buffer));
+        ackStream.onCompleted();
+      }
+    };
+  }
+
+  @Override
+  public Runnable acceptPutStatement(FlightSql.CommandStatementUpdate command, CallContext context,
+                                     FlightStream flightStream,
+                                     StreamListener<PutResult> ackStream) {
+    IntegrationAssertions.assertEquals(command.getQuery(), "UPDATE STATEMENT");
+
+    return acceptPutReturnConstant(ackStream, 10000);
+  }
+
+  @Override
+  public Runnable acceptPutPreparedStatementUpdate(FlightSql.CommandPreparedStatementUpdate command,
+                                                   CallContext context, FlightStream flightStream,
+                                                   StreamListener<PutResult> ackStream) {
+    IntegrationAssertions.assertEquals(command.getPreparedStatementHandle().toStringUtf8(),
+        "UPDATE PREPARED STATEMENT HANDLE");
+
+    return acceptPutReturnConstant(ackStream, 20000);
+  }
+
+  @Override
+  public Runnable acceptPutPreparedStatementQuery(FlightSql.CommandPreparedStatementQuery command,
+                                                  CallContext context, FlightStream flightStream,
+                                                  StreamListener<PutResult> ackStream) {
+    IntegrationAssertions.assertEquals(command.getPreparedStatementHandle(),
+        "SELECT PREPARED STATEMENT HANDLE");
+
+    return null;
```

Comment:
Hmm, this should throw an exception if it's not implemented.

Reply:
Implemented this case as well